### ISSUE TITLE
Issue of generate the final FASTA file after manual curation in Juicebox

### ISSUE THREAD

**Issue by haoran-chn** [zengxiaofei/HapHiC#36]
Dear Xiaofei,

Thanks for providing this useful tool. I assembled the genome (2n=16, 8 chromosomes) by HiFi reads only using hifiasm. 

stats for asm.bp.p_ctg.fasta
sum = 255015880, n = 387, ave = 658955.76, largest = 30935753
N50 = 11339244, n = 7
N60 = 10070553, n = 9
N70 = 9531125, n = 12
N80 = 7505767, n = 15
N90 = 4309235, n = 19
N100 = 20579, n = 387
N_count = 0
Gaps = 0

Then I run HapHiC with one-line command: `/path/to/HapHiC/haphic pipeline asm.bp.p_ctg.fasta HiC.filtered.bam 8`
and then I use `$bash juicebox.sh` to generate the .hic and .assembly for manual curation.
after manual curation in Juicebox, I run `/path/to/HapHiC/utils/juicer post -o out_JBAT out_JBAT.review.assembly out_JBAT.liftover.agp asm.bp.p_ctg.fasta` to generate the final FASTA file.

**I found issue of the FINAL FASTA file.**
stats for out_JBAT.FINAL.fa
sum = 372880511, n = 9, ave = 41431167.89, largest = 136096530
N50 = 33899492, n = 3
N60 = 30935753, n = 4
N70 = 30117002, n = 5
N80 = 29705702, n = 7
N90 = 24223963, n = 8
N100 = 23588956, n = 9
N_count = 38700
Gaps = 387

**I do not know why the total bases increased a lot. The size for scaffolds 1-8 (~30 Mb) looks good according to the final heatmap (attachment), but there is a huge scaffold 9 which is 136 Mb. I do not know how this scaffold 9 was created. Do you know if the scaffold 1-8 sequence is correct?**
[contact_map.pdf](https://github.com/user-attachments/files/16353545/contact_map.pdf)
[separate_plots.pdf](https://github.com/user-attachments/files/16353546/separate_plots.pdf)

I do not have any problem when running the pipeline, no error report. I also made the plot use out_JBAT.FINAL.agp, it also looks good to me.

Thanks,
Haoran

![FINAL-FASTA](https://github.com/user-attachments/assets/1016194d-a9d3-4c81-aaef-c54da2a1f752)

**Comment by zengxiaofei** [comment]
Hi Haoran,

Could you please provide your `out_JBAT.FINAL.agp`, `out_JBAT.review.assembly`, and `out_JBAT.liftover.agp`? Thank you.

Best regards,
Xiaofei

**Comment by haoran-chn** [comment]
Hi Xiaofei,

I redraw the plot use out_JBAT.FINAL.agp and HiC.filtered.bam, it actually has the huge scaffold 9.
Here I provide the file you required.

[contact_map2.pdf](https://github.com/user-attachments/files/16359319/contact_map2.pdf)
[file.zip](https://github.com/user-attachments/files/16359326/file.zip)


Best,
Haoran

**Comment by zengxiaofei** [comment]
Hi Haoran,

I have checked the `out_JBAT.FINAL.agp` and `out_JBAT.review.assembly` files you provided. It seems that you combined all un-anchored contigs into a single scaffold in Juicebox. Although this approach is also acceptable and commonly used (known as 'chrUN'), I think it is a little bit misleading. To address this, you can correct this issue in Juicebox by using the `Add chr boundaries` command, and then generate a new `.review.assembly` file.

Another point worth mentioning is that there may be a high level of heterozygosity in your genome, as evident by many redundant regions on the contact map:

![group2](https://github.com/user-attachments/assets/1fdcf386-14b7-4eab-8d20-2e455efb95d3)

To verify this, you can perform self-alignment or identify duplicated BUSCOs. It is advisable to further purge duplications to achieve a haplotype-collapsed primary assembly.

Best wishes,
Xiaofei

**Comment by haoran-chn** [comment]
Hi Xiaofei,

Thanks for your response. Yes, combined all un-anchored contigs into a single scaffold. But how does this increase the total base from 255015880 into 372880511 (asm.bp.p_ctg.fasta vs out_JBAT.FINAL.fa)?

I also feel curious about the high level of heterozygosity, cause the GenomeScope does not show the high ratio.
![linear_plot2](https://github.com/user-attachments/assets/f5faee62-13c6-4997-8116-93fe5d82b847)

Best,
Haoran

**Comment by zengxiaofei** [comment]
`out_JBAT.liftover.agp` includes 387 contigs with a total length of 255,015,880 bp, while `out_JBAT.FINAL.agp` contains 396 contigs with a total length of 372,841,811 bp. These replicated contigs are ptg000002l, ptg000007l, ptg000006l, ptg000014l, ptg000010l, ptg000003l, ptg000019l, and ptg000020l.

Then I checked the `out_JBAT.review.assembly` file. Finally, I found that there is additional space in the last line of this file. After removing this space, `juicer post` can generate correct files.

> I also feel curious about the high level of heterozygosity, cause the GenomeScope does not show the high ratio.

The histograms output by hifiasm are more valuable.
